examples/asyncWorkflow/main.go

```diff
@@ -0,0 +1,151 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"time"
+
+	dbsql "github.com/databricks/databricks-sql-go"
+	dbsqlctx "github.com/databricks/databricks-sql-go/driverctx"
+	dbsqllog "github.com/databricks/databricks-sql-go/logger"
+	"github.com/joho/godotenv"
+)
+
+func main() {
+	// use this package to set up logging. By default logging level is `warn`. If you want to disable logging, use `disabled`
+	if err := dbsqllog.SetLogLevel("debug"); err != nil {
+		panic(err)
+	}
+	// sets the logging output. By default it will use os.Stderr. If running in terminal, it will use ConsoleWriter to make it pretty
+	// dbsqllog.SetLogOutput(os.Stdout)
+
+	// this is just to make it easy to load all variables
+	if err := godotenv.Load(); err != nil {
+		panic(err)
+	}
+	port, err := strconv.Atoi(os.Getenv("DATABRICKS_PORT"))
+	if err != nil {
+		panic(err)
+	}
+
+	// programmatically initializes the connector
+	// another way is to use a DNS. In this case the equivalent DNS would be:
+	// "token:<my_token>@hostname:port/http_path?catalog=hive_metastore&schema=default&timeout=60&maxRows=10&&timezone=America/Sao_Paulo&ANSI_MODE=true"
+	connector, err := dbsql.NewConnector(
+		// minimum configuration
+		dbsql.WithServerHostname(os.Getenv("DATABRICKS_HOST")),
+		dbsql.WithPort(port),
+		dbsql.WithHTTPPath(os.Getenv("DATABRICKS_HTTPPATH")),
+		dbsql.WithAccessToken(os.Getenv("DATABRICKS_ACCESSTOKEN")),
+		//optional configuration
+		dbsql.WithUserAgentEntry("workflow-example"),
+		dbsql.WithInitialNamespace("hive_metastore", "default"),
+		dbsql.WithTimeout(time.Minute), // defaults to no timeout. Global timeout. Any query will be canceled if taking more than this time.
+		dbsql.WithMaxRows(10),          // defaults to 10000
+	)
+	if err != nil {
+		// This will not be a connection error, but a DSN parse error or
+		// another initialization error.
+		panic(err)
+
+	}
+	// Opening a driver typically will not attempt to connect to the database.
+	db := dbsql.OpenDB(connector)
+	// make sure to close it later
+	defer db.Close()
+
+	db.SetMaxOpenConns(2)
+
+	// the "github.com/databricks/databricks-sql-go/driverctx" has some functions to help set the context for the driver
+	ogCtx := dbsqlctx.NewContextWithCorrelationId(context.Background(), "asyncWorkflow-example")
+
+	// for _, v := range []string{"1", "2", "3", "4", "5", "6", "7", "8", "9", "10"} {
+	// i := v
+	// go func() {
+	// _, exc, err := db.QueryContext(ogCtx, fmt.Sprintf("select %s", i))
+	rs, exc, err := db.QueryContext(ogCtx, `SELECT id FROM RANGE(100) ORDER BY RANDOM() + 2 asc`)
```

Comment:
To make sure I'm understanding the semantics of this correctly:

1. If the query has completed (direct results), then `rs` will be set and the caller can fetch the results
2. Otherwise, the caller can use `exc` to get the operation ID(?), and use that to poll for query status?

Reply:
hi @aldld , that's the idea... there is more to the execution than the ID, but it is just data that can be saved and used later.
I am not very happy with the way to check the results yet with different methods to query, check, get results. There must be a simpler way to merge check and get results in the same call. See the example I added so you have an idea of what I am talking about.